### DISCUSSION TITLE
Font rendering issue with some languages in aarch64 version. (CJK font not work on Linux arm64)

### DISCUSSION THREAD

**Discussion by ambitiousCC** [rustdesk/rustdesk#7908] [General]
### Bug Description

1. When attempting to run the aarch64 version (including both AppImage and .deb installation methods) on the third-party Linux operating system KylinOS-arm64, font rendering issues occur with Chinese, Japanese, and other languages. However, similar issues are not observed in the x86_64 version of the same system. All normal functionalities seem to work properly. Additionally, it's noticed that there are differences in the output information between the two versions. Could this be a software compatibility issue?
* kylinOS - arm64
![err1](https://github.com/rustdesk/rustdesk/assets/45097033/36fd2e52-9bbc-4441-9442-7e0820128c00)
* kylinOS - x86_64 - The first run generates additional output related to font configuration.
![succ2](https://github.com/rustdesk/rustdesk/assets/45097033/4bddf787-6d78-4443-ae95-24067ace584e)
2. When compiled and installed according to the project's instructions, running the application eventually results in a segmentation fault. Even in the Debug version, no crucial information is provided to further identify and address the issue. However, it's worth noting that the release version runs without any problems. Could this be due to incompatibility issues with the latest version of the code?
![unknown](https://github.com/rustdesk/rustdesk/assets/45097033/297faaa1-8646-4bc5-b2ec-cba3f8f95bb9)

### How to Reproduce

You can reproduce this issue by installing the ARM version of the system. [Download link](https://distro-images.kylinos.cn:8802/web_pungi/download/share/M8UbGlg2WyeHnANzv0srJOEjC9R7ZXDx/Kylin-Desktop-V10-SP1-General-Release-2303-ARM64.iso.torrent)

### Expected Behavior

The font display issue across various languages has been fixed, and we've also addressed the issue of the local code not running properly.

### Operating system(s) on local side and remote side

all -> Linux - arm64 

### RustDesk Version(s) on local side and remote side

all -> all

### Screenshots

* x86其他语言也能够正常显示
![20240505134739](https://github.com/rustdesk/rustdesk/assets/45097033/59fd02dc-d132-48fa-a99a-9d3b70cdc081)

### Additional Context

These issues might also stem from system-related issues, but there are unclear aspects on the software side as well. Let's discuss and see if we can find a solution.

**Comment by xlionjuan** [comment]
Try 1.2.4 nightly

**Reply by ambitiousCC** [reply]
Using version 1.2.4-nightly, it's discovered that the application fails to run, with the error message: "APPRUN_ERROR: No such file or directory." Compiling from source code also results in the same issue as before.
<img width="182" alt="1 2 4 bug" src="https://github.com/rustdesk/rustdesk/assets/45097033/7a3d1db0-0658-4b7d-ac01-088623cee82e">

**Reply by xlionjuan** [reply]
If I didn't remember wrong, your system is based on Debian or Ubuntu, please install `.deb` instead of appimage.

**Reply by ambitiousCC** [reply]
The application now runs successfully using the .deb package. However, some languages still appear as garbled characters. The languages affected are:
1. Arabic (ar)
2. Persian (fa)
3. Hebrew (he)
4. Japanese (ja)
5. Korean (ko)
6. Thai (th)
7. Chinese (Simplified) (zh-cn)
8. Chinese (Traditional) (zh-tw)
I wonder if there's any commonality among these languages that causes them to not display properly.

**Reply by ambitiousCC** [reply]
I tried on another Linux system, [UOS](https://cdimage-download.chinauos.com/desktop-professional/1060u3/uos-desktop-20-professional-1060-arm64-202401-fix.iso?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE3MTQ5Nzc3NDQsImZpbGVfbmFtZSI6ImRlc2t0b3AtcHJvZmVzc2lvbmFsLzEwNjB1My91b3MtZGVza3RvcC0yMC1wcm9mZXNzaW9uYWwtMTA2MC1hcm02NC0yMDI0MDEtZml4LmlzbyJ9.MInl-c1FmW0sfiOWb6dXVRUe0riClfX0V0D5A1Xzun0), and found that the garbled character issue persists, but fewer fonts are affected. It only includes:

    Japanese (ja)
    Korean (ko)
    Thai (th)
    Chinese (Simplified) (zh-cn)
    Chinese (Traditional) (zh-tw)

**Reply by ambitiousCC** [reply]
Oops! The garbled character issue also occurs on Ubuntu 20.04, similar to Kylin. Currently, I suspect it might be due to running the virtual machine on PVE.
<img width="719" alt="ubuntu2004" src="https://github.com/rustdesk/rustdesk/assets/45097033/bd8592fb-a710-49a8-a81b-2e88f2dd0abe">

**Reply by ambitiousCC** [reply]
Testing on a physical machine is complete, and the same garbled character issue has occurred.

**Reply by ambitiousCC** [reply]
Could it be due to the inconsistency in Flutter framework support between x86_64 and arm64 architectures?

**Reply by lzjzjw** [reply]
Hello, author. May I ask if your issue has been resolved. Is there a solution?

**Reply by kairenxue** [reply]
Hello, author. May I ask if your issue has been resolved. Is there a solution?

**Comment by rustdesk** [comment]
https://github.com/flutter/flutter/issues/139293#issuecomment-2115345709 @21pages

**Reply by rustdesk** [reply]
https://github.com/notofonts/noto-cjk/releases these files are big, it not feasible to put these files in the build.

**Reply by kairenxue** [reply]
Could the problem be solved by copying the font file to the font directory of RustDesk and specifying the font loading path when starting up?

**Comment by xlionjuan** [comment]
Ubuntu 24's new App Center is also using Flutter, and also broken 😂

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/issues/10666
